### PR TITLE
[CARBONDATA-2010] Block streaming on main table of preaggregate datamap

### PR DESCRIPTION
**Pull request by QiangCai**
If the table has 'preaggregate' DataMap, it doesn't support streaming now

 - [x] Any interfaces changed?
   no
 - [x] Any backward compatibility impacted?
   no
 - [x] Document update required?
   yes, i will add this limitation into
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
          added new ut
        - How it is tested? Please attach test report.
           CI run ut
        - Is it a performance related change? Please attach the performance test report.
           no
        - Any additional information to help reviewers in testing this change.
           added code comment
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
  small changes

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2695/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1462/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2832/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2714/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1481/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2841/

File: core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableInfo.java [hunk 1]
```diff
@@ -316,4 +316,18 @@ public static TableInfo deserialize(byte[] bytes) throws IOException {
     return parentRelationIdentifiers;
   }
 
+  /**
+   * whether this table has aggregation DataMap or not
+   */
+  public boolean hasAggregationDataMap() {
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableInfo.java]: I think it is ok you move this logic to CarbonTable.hasAggregationDataMap directly. We better keep TableInfo unmodified

Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableInfo.java]: ok

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1514/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2746/

**Comment by QiangCai** [comment]
retest sdv please

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2774/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1541/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**